server/events_test.go

```diff
@@ -1107,3 +1107,40 @@ func TestServerEventStatsZ(t *testing.T) {
 		t.Fatalf("Expected a route, but got %d", lr)
 	}
 }
+
+func TestServerEventsPingStatsZ(t *testing.T) {
+	sa, _, sb, optsB, akp := runTrustedCluster(t)
+	defer sa.Shutdown()
+	defer sb.Shutdown()
+
+	url := fmt.Sprintf("nats://%s:%d", optsB.Host, optsB.Port)
+	nc, err := nats.Connect(url, createUserCreds(t, sb, akp))
+	if err != nil {
+		t.Fatalf("Error on connect: %v", err)
+	}
+	defer nc.Close()
+
+	reply := nc.NewRespInbox()
+	sub, _ := nc.SubscribeSync(reply)
+
+	nc.PublishRequest(serverStatsPingReqSubj, reply, nil)
+
+	// Make sure its a statsz
+	m := ServerStatsMsg{}
+
+	// Receive both manually.
+	msg, err := sub.NextMsg(randomBackoff)
+	if err != nil {
+		t.Fatalf("Error receiving msg: %v", err)
+	}
+	if err := json.Unmarshal(msg.Data, &m); err != nil {
+		t.Fatalf("Error unmarshalling the statz json: %v", err)
+	}
+	msg, err = sub.NextMsg(randomBackoff)
```

Comment:
Same than above. I would do something like 2*randomBackoff. And again, this is timeout, does not mean that this call will block for that long.

Reply:
You are right about this one.